### PR TITLE
 feat(lambda-tiler): start serving tile json

### PR DESCRIPTION
**Pull request by Wentao-Kuang**

### CODE REVIEW

File: packages/lambda-tiler/src/routes/tile.ts [hunk 1]
```diff
@@ -117,12 +117,43 @@ export async function wmts(req: LambdaContext): Promise<LambdaHttpResponse> {
     return response;
 }
 
+export async function tileJson(req: LambdaContext): Promise<LambdaHttpResponse> {
+    const { rest } = req.action;
+    const host = Env.get(Env.PublicUrlBase) ?? '';
+    const tileUrl = `${host}/${rest[0]}/${rest[1]}/{z}/{x}/{y}.pbf`;
+
+    const tileJson = {
```
Comment (blacha) [inline packages/lambda-tiler/src/routes/tile.ts:125]: could you please define a TileJson interface

File: packages/lambda-tiler/src/routes/tile.ts [hunk 2]
```diff
@@ -117,12 +117,43 @@ export async function wmts(req: LambdaContext): Promise<LambdaHttpResponse> {
...
+    const data = Buffer.from(json);
+
+    const cacheKey = createHash('sha256').update(data).digest('base64');
+
+    const respNotMod = checkNotModified(req, cacheKey);
+    if (respNotMod != null) return respNotMod;
+
+    const response = new LambdaHttpResponse(200, 'ok');
+    response.header(HttpHeader.ETag, cacheKey);
+    response.header(HttpHeader.CacheControl, 'max-age=0');
```
Comment (blacha) [inline packages/lambda-tiler/src/routes/tile.ts:144]: I think we want to cache this for a long time, so might need to revist this.

File: packages/lambda-tiler/src/__test__/xyz.test.ts [hunk 3]
```diff
@@ -198,6 +198,46 @@ o.spec('LambdaXyz', () => {
...
+        });
+
+        o('should 304 if a json is not modified', async () => {
+            const key = 'fYC3MHM69e9U9ltWQLry45/G1wMjTeauij8L2HUdVk4=';
+            const request = mockRequest('/v1/tiles/tile.json', 'get', { 'if-none-match': key });
+
+            const res = await handleRequest(request);
+            if (res.status === 200) {
+                o(res.header('eTaG')).equals(key); // this line is useful for discovering the new etag
+                return;
```
Comment (blacha) [inline packages/lambda-tiler/src/__test__/xyz.test.ts:214]: do we want to throw if the status is 200?

File: packages/lambda-tiler/src/__test__/xyz.test.ts [hunk 4]
```diff
@@ -198,6 +198,46 @@ o.spec('LambdaXyz', () => {
...
+            const request = mockRequest('/v1/tiles/topolike/Google/tile.json');
+            request.apiKey = 'secretKey';
+
+            const res = await handleRequest(request);
+            o(res.status).equals(200);
+            o(res.header('content-type')).equals('application/json');
+            o(res.header('cache-control')).equals('max-age=0');
+
+            const body = Buffer.from(res.getBody() ?? '', 'base64').toString();
+            o(body).equals(
```
Comment (blacha) [inline packages/lambda-tiler/src/__test__/xyz.test.ts:235]: would be nicer to convert this into a `deepEquals`  the actual object

File: packages/lambda-tiler/src/__test__/xyz.test.ts (hunk 3 above)
Comment (Wentao-Kuang) [reply packages/lambda-tiler/src/__test__/xyz.test.ts:214]: This is used to discovering the if the etag changes. I don't think tilejson is immutable. We still want to update the layers.

Comment (blacha) [reply packages/lambda-tiler/src/__test__/xyz.test.ts:214]: does the test fail if `status == 200`

**Review by blacha — APPROVED** [review]

**Pull request merged**